### PR TITLE
Refactor status handling in ASReview backend

### PR DESCRIPTION
**Pull request by J535D165**

### CODE REVIEW

File: asreview/webapp/api.py [hunk 1]
```diff
@@ -250,32 +251,15 @@ def api_get_project_info(project):  # noqa: F401
...
     project_config = project.config
 
-    try:
-
-        # check if there is a dataset
-        try:
-            get_data_file_path(project.project_path)
-            project_config["projectHasDataset"] = True
-        except Exception:
-            project_config["projectHasDataset"] = False
```
Comment (terrymyc) [inline asreview/webapp/api.py:260]: This `projectHasDataset` property is needed in the frontend.

Comment (J535D165) [reply asreview/webapp/api.py:260]: Yes, thanks. Please check the project info for `dataset_path`. If not exists or is null, then there is no dataset. Does that help?

**Review by terrymyc — COMMENTED** [review]
There are some places in `api.py` where `projectInitReady` is not deprecated yet.
https://github.com/asreview/asreview/pull/1011/files#diff-2d1d7df27b25eb5f202ef7ce660d1f86940bc78bc2936b70728a1c415862cb5cR120-R122

https://github.com/asreview/asreview/pull/1011/files#diff-2d1d7df27b25eb5f202ef7ce660d1f86940bc78bc2936b70728a1c415862cb5cR161-R173

**Comment by J535D165** [comment]
Oke thanks, I will have a look!

**Comment by J535D165** [comment]
I resolved the issues in commit 9be8c034c9c25e6f62fb0abfb92e3c88c8e3ab9d. Also made some fixes for handling the status of projects without datasets (they don't have a status atm).

**Comment by J535D165** [comment]
Thanks a lot for your help @terrymyc

**Comment by terrymyc** [comment]
You are welcome! @J535D165

**Pull request merged**